_javascript/modules/components/toc/toc-mobile.js

```diff
@@ -62,7 +62,9 @@ export class TocMobile {
     activeItem.scrollIntoView({ block: 'center' });
   }
 
-  static hidePopup() {
+  static hidePopup(event) {
+    event?.preventDefault();
+
```

Comment:
This doesn't make it clear what the intent of this line of code is, how about adding a specific condition `event.type === 'cancel'`?

```suggestion
  static hidePopup(event) {
    if (!$popup.open) {
      return;
    }

    if (event && event.type === 'cancel') {
      event.preventDefault();
    }
```